jlh/JonFirstApp/app/src/main/java/com/e_conomic/jonfirstapp/MainActivity.java

```diff
@@ -96,11 +118,37 @@ public void sendMessage(String message) {
         // If the display message fragment is visible show the message there.
         if (displayMessageFragment != null) {
             displayMessageFragment.updateMessage(message);
-            return;
+        } else {
+            Log.w(MAIN_ACTIVITY_TAG,
+                    "Trying to send message, but displayMessageFragment is null.");
+        }
+
+        if (messageFile == null) {
```

Comment:
Yeah, I understand we have to move the check to `writeMessageToFile`and guard against the exceptional case that the file does not exist, **but** is the check `messageFile == null` not redundant here? The code in `writeMessageToFile` doesn't depend on the `messageFile` reference (it just uses to filename to create an output stream).

As I'm looking into it, it actually looks like [openFileOutput](https://developer.android.com/reference/android/content/Context.html#openFileOutput(java.lang.String, int)) creates a file if it doesn't already exist, so that the `messageFile` reference is not needed at all? 

Tell me if I'm just confused and looking at it completely in the wrong way, but it seems like the `openFileOutput` handles all our problems for us?